### DISCUSSION TITLE
Airflow Web Server crashes on sqlalchemy.py

### DISCUSSION THREAD

**Discussion by ItayKirgnerVerix** [apache/airflow#20406] [Q&A]
We are using Amazon Managed Workflows for Apache Airflow (MWAA) and we noticed that suddenly after some Variable creation and minor DAG update we can't access the Apache Airflow UI and getting the error message “ERR_EMPTY_RESPONSE” in the UI web page and while looking at the webserver console log we see the error:

Log stream name: webserver_console_ip-xx-x-xx-xxx.ec2.internal_xxxxxxxxxx.xxxxxxx.log
In script: `sqlalchemy.py`
In function: `process_result_value`
In line 176 `"return type_map[data['type']](**data['attrs'])"`
Error message:  `KeyError: 'type'`

The Error chain starts from “File "/usr/local/lib64/python3.7/site-packages/sqlalchemy/util/compat.py", line 178, in raise_”

Looking at the logs, the webserver tries to boot itself up and crashes repeatedly in the same line of code.

Important to note that we see that the environment is still working as some DAGs (but not all) are still completing their tasks (we are receiving emails of confirmation).

Before the error occurred, we created a new variable and used it in a DAG, I don’t believe this should affect the Web Server… The site should start normally even in a case where a DAG is "broken". 
We have not made any changes to the environment other than that.

Logs of the error:
![image](https://user-images.githubusercontent.com/85482434/146731952-9767b069-9a08-4768-8beb-85da71498b1c.png)

**Comment by potiuk** [comment]
This is a very strange error - I think you should raise it to MWAA support (if you are using MWAA) or talk to your devops because it really looks like a low-level problem of configuring your database. Looks like your webserver is not communicating with the right database or maybe your driver is broken somehow or maybe you have some specific database deployment that causes it. That's the first time ever I see this kind of problem and it's almost for sure deployment related.

**Comment by lcruz-coursera** [comment]
@ItayKirgnerVerix I'm having the same error. Was there any resolution on this?

**Reply by ItayKirgnerVerix** [reply]
Hi, I don't remember exactly what happened, but the cause of it was the variable we created in Airflow and used in the DAG's configuration. This was problematic as it created dependency issues - if I remember correctly the DAG is loaded before the variables. So you can't use them when configurating the DAG.

I believe we resolved it by creating a new environment, and without using the new variable. Not the most ideal solution, but we couldn't access the variables at all due to the issue crashing the webserver every time.
